src/test/codegen/enum-bounds-check.rs

```diff
@@ -3,12 +3,20 @@
 #![crate_type = "lib"]
 
 pub enum Foo {
-    A, B
+    A = 2,
+    B = 3
 }
 
 // CHECK-LABEL: @lookup
 #[no_mangle]
-pub fn lookup(buf: &[u8; 2], f: Foo) -> u8 {
+pub fn lookup(buf: &[u8; 5], f: Foo) -> u8 {
     // CHECK-NOT: panic_bounds_check
-    buf[f as usize]
+    buf[f as usize + 1]
+}
```

Comment:
Let's also make sure that we still test for the previous case where we use `f as usize` without incrementing or decrementing it.

Reply:
Certainly, I restored it. However, there's still the problem that the assumptions may or may not have any effect, so right now the newer test fail at least when tested with LLVM8 :(

Reply:
That's inconvenient. I'm guessing we _could_ filter the test to only be run on LLVM10, but I'll defer to someone else for the determination if that is acceptable. Then again, this is an optimization that we currently don't have, so it might make sense to do so.